docs/DeveloperGuide.adoc

```diff
@@ -12,255 +12,359 @@ ifdef::env-github[]
+* <<Design-SuggestionEngine,*`SuggestionEngine`*>>: Deals with suggestions generation.
+* <<Design-CorrectionEngine,*`CorrectionEngine`*>>: Deals with auto-correction.
 
-.Interactions Inside the Logic Component for the `delete 1` Command
+[[Design-NotablyParser]]
+==== NotabyParser component
+
+.Class Diagram of the Logic Component
+image::LogicClassDiagram.png[]
+
+.  `Logic` uses the `NotablyParser` class to parse the user command.
+.  This results in a `List<Command>` object which is executed by the `LogicManager`.
+.  The command execution can affect the `Model` (e.g. adding a Note).
+.  The updated model/data structure will automatically be reflected on to the `View`.
+
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete cs2103")` API call.
+
+.Interactions Inside the Logic Component for the `delete -t cs2103` Command
 image::DeleteSequenceDiagram.png[]
 
 NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
 
+[[Design-SuggestionEngine]]
+==== SuggestionEngine component
+
+.Class Diagram of the Suggestion Engine Component (still WIP)
+image::SuggestionClassDiagram.png[]
+
+*API* :
+link:{repoURL}blob/master/src/main/java/com/notably/logic/suggestion/SuggestionEngine.java[`SuggestionEngine.java`]
+
+`SuggestionEngine` gives users the meaning of the command they input and a list of notes suggestions that they want to
+open/ delete/ search/ edit.
+
+1. `Logic` uses the `SuggestionEngineImpl` class, which implements `SuggestionEngine` interface, to parse the user command.
+2. According to the command the user inputs, `SuggestionEngineImpl` will create a `XYZSuggestionCommandParser` object
+which implements `SuggestionCommandParser` interface.
+3. This results in the creation of `XYZSuggestionCommand` object which implements `SuggestionCommand` interface.
+`XYZSuggestionCommand` is then executed by the `LogicManager`.
+4. The command execution can affect the `Model` (e.g. `open` command will set the response text in the model as
+"Open a note" and store a list of `SuggestionItem` in the model).
+5. The UI will then be able to retrieve the response text and list of `SuggestionItem` from the model to be displayed
+to the user.
+
+Given below is the Sequence Diagram for interactions within the `Logic` and `Suggestion` component for the `execute("open /")` API call.
+
+.Interactions Inside the Logic and Suggestion Component for the `open /` Command
+image::SuggestionSequenceDiagram.png[]
+
+NOTE: The lifeline for `OpenSuggestionCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+
+[[Design-CorrectionEngine]]
+==== CorrectionEngine component
+
+.Class Diagram of the CorrectionEngine Component
+image::CorrectionEngineClassDiagram.png[]
```

Comment:
The interactions between CorrectionEngine to CorrectionStatus and CorrectionResult are not explained and cannot be inferred from the diagram. Maybe some clarification would be useful for developers to understand this.